news: Nix development environment and package (#440)
OSGeo/grass-website: content/news/2024_06_07_Nix_environment_and_package_added.md

---
title: "Nix development environment and package"
date: 2024-07-06T12:00:00+02:00
layout: "news"
author: Ivan Mincik and the GRASS Dev Team
---

> ### You can now develop and run GRASS with Nix

Reviewer: ```suggestion
### You can now develop and run GRASS GIS with Nix
```

Reviewer: (at least in the section titles...)

---
title: "Nix development environment and package"
date: 2024-07-06T12:00:00+02:00
layout: "news"
author: Ivan Mincik and Veronica Andreo
---

## Nix development environment and package

New option for creating a GRASS development environment and unique way
of running GRASS directly from a Git source code was implemented using
[Nix package manager](https://nixos.org/). This idea was presented by
[Ivan Mincik](https://github.com/imincik)
during a
[GRASS GIS Community Meeting](https://grasswiki.osgeo.org/wiki/GRASS_Community_Meeting_Prague_2024)
in Prague.

> Nix development environment provides a stable and reproducible environment

Reviewer 2: ```suggestion
The Nix development environment provides a stable and reproducible environment
```

> for all developers and can significantly simplify onboarding process of new

Reviewer 2: ```suggestion
for all developers and can significantly simplify the onboarding process of new
```

contributors.

> To start hacking on GRASS, run `nix develop` command in source code root

Reviewer 2: ```suggestion
To start hacking on GRASS, run `nix develop` command within the source code root
```

directory to create a temporary shell environment containing all build

> dependencies with instructions how to build and test the code.

Reviewer 2: ```suggestion
dependencies and get instructions on how to build and test the code.
```

```
$ nix develop

Welcome to a GRASS development environment !
Build GRASS using following commands:

 1.  ./configure --prefix=$(pwd)/app
 2.  make -j16
 3.  make install

Run tests:

 1.  export PYTHONPATH=$(app/bin/grass --config python_path):$PYTHONPATH
 2.  export LD_LIBRARY_PATH=$(app/bin/grass --config path)/lib:$LD_LIBRARY_PATH
 3.  pytest

Note: run 'nix flake update' from time to time to update dependencies.

Run 'dev-help' to see this message again. 
```  

> On the other hand, possibility to run GRASS directly from a Git

Reviewer 2: ```suggestion
On the other hand, the possibility to run GRASS directly from the Git
```

> source code improves feedback loop during development and will allow to install
> GRASS on any Linux distribution immediately when new version is out.

Reviewer 2: ```suggestion
source code improves the feedback loop during development and will allow to install
GRASS on any Linux distribution immediately after a new version is out.
```

> To run latest development version of GRASS directly from a Git source code, run

Reviewer 2: ```suggestion
To run the latest development version of GRASS directly from the Git source code, run
```

`nix run github:OSGeo/grass#grass` command which will compile (if needed) and
run GRASS in one step.

<a href="/images/news/nix-run-grass.png">
  <img src="/images/news/nix-run-grass.png" alt="nix run grass" title="Nix running GRASS"

>    width="70%" style="float:center;padding-left:10px;padding-top:20px">

Reviewer 2: ```suggestion
   width="90%" style="float:center;padding-left:10px;padding-top:20px">
```

</a>

> To install it, run `nix profile install github:OSGeo/grass#grass`.

Reviewer 2: ```suggestion
If you want to install GRASS instead, run `nix profile install github:OSGeo/grass#grass`.
```

Check out [How to use Nix](https://github.com/OSGeo/grass/blob/main/NIX.md) for more instructions.

> *Ivan Mincik and Veronica Andreo*

Reviewer 2: ```suggestion
*Ivan Mincik*
```
